students/randi_peterson/session03/mailroom.py

```diff
@@ -0,0 +1,77 @@
+def ThankYou(donorlst):
+    fullname = input('What is the full name of the donor? Type list to see donors. Type exit to quit task \n')
+    if fullname.lower() == 'list':
+        for donor in donorlst:
+            print(donor[0])
+    elif fullname.lower() == "exit":
+        return None
+    else:
+        #Find where donor is in list or create a new one
+        donorindex = None
+        for index,donors in enumerate(donorlst):
+            if donors[0] == fullname:
+                donorindex = index
+                break
+        if donorindex == None:
+            donorindex = len(donorlst)
+            new_entry = [fullname]
+            donorlst.append(new_entry)
+        newdonation = float(input('What is the value of the donation (numbers only)?\n'))
+        donorlst[donorindex].append(newdonation)
+
+        #Print the thank you email
+        print(f'Dear {fullname}, \n Thank you for your generous donation. We appreciate the support from people like you. \n Thank you,\n Charity Name')
+
+def CreateReport(donorslist):
+    # Create report data from donor list
+    j = 0
+    donor_report = []
+    for donor in donorlst:
+        name = donorlst[j][0]
+        donations = len(donorlst[j]) - 1
+        total_given = 0
+        # Add up all donations
+        k = 1
+        while k < donations + 1:
+            total_given += donorlst[j][k]
+            k += 1
+        avg_gift = total_given / donations
+        donor_report.append([name, total_given, donations, avg_gift])
+        j += 1
```

Comment:
you don't need manual counter - you are looping over the donor list so `donor` is your list item, use it.